### PR TITLE
Support Data Driven Testing

### PR DESCRIPTION
**Pull request by APshenkin**
Our goal: https://github.com/Codeception/CodeceptJS/issues/239

TO-DO list:
- [x] Add DDT support for Scenario
- [x] Do not allow to use ScenarioContext as a PageObject
- [x] Replace "bad" characters in screenshot name (like " in Windows) https://github.com/Codeception/CodeceptJS/issues/426
- [x] Write unit test
- [ ] Add documentation
- [x] Create datatable for Feature/suite
- [x] do not breaks when scenario.only

Status of current realization:
```
Feature('@smoke tests');

Before((I) => {
    I.amOnPage("/");
});

let accounts = new dataTable(['login', 'password']); <- new class, that will store dataTable
dataTableb.add(['davert', '123456']); <- adding  records to table
dataTableb.add(['admin', '123456']);


# You have to pass dataTable to Scenario
# You have to add special `scenarioContext` to function params 
Scenario('Test some crazy', accounts, (I, scenarioContext) => {
  I.fillField('Username', scenarioContext.login); <- access to scenarioContext
  I.fillField('Password', scenarioContext.password);
  I.click('Sign In');
  I.see('Welcome '+ scenarioContext.login);
});


output
@smoke tests --
 Test some crazy {"login":"davert","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "davert"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Welcome davert"
 ✓ OK in 2468ms

 Test some crazy {"login":"admin","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "admin"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Welcome admin"
 ✓ OK in 2468ms


Also you can set datatable to Feature. Then All tests in this Feature will be launched several times with different data. Also you can override this table in Scenario

Example:
let accounts1 = new DataTable(['login', 'password']);
accounts1.add(['davert', '123456']);
accounts1.add(['admin', '123456']);


Feature('@smoke Тесты футера', accounts1);

let accounts2 = new DataTable(['login', 'password']);
accounts2.add(['andrey', '123456']);
accounts2.add(['collaborator', '123456']);

Before((I, mainPage) => {
    I.amOnPage("/about");
});

Scenario('Test some crazy', (I, scenarioContext) => {
    I.fillField('Username', scenarioContext.login); <- access to scenarioContext
    I.fillField('Password', scenarioContext.password);
    I.click('Sign In');
    I.see('Welcome '+ scenarioContext.login);
});

Scenario('Test some crazy with accounts2', accounts2, (I, scenarioContext) => {
    I.fillField('Username', scenarioContext.login); <- access to scenarioContext
    I.fillField('Password', scenarioContext.password);
    I.click('Sign In');
    I.see('Access for user '+ scenarioContext.login + ' is denied.');
});

output
@smoke tests --
 Test some crazy {"login":"davert","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "davert"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Welcome davert"
 ✓ OK in 2468ms

 Test some crazy {"login":"admin","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "admin"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Welcome admin"
 ✓ OK in 2468ms

 Test some crazy with accounts2 {"login":"andrey","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "andrey"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Access for user andrey is denied.'"
 ✓ OK in 2468ms

 Test some crazy with accounts2 {"login":"collaborator","password":"123456"}
 • I am on page "/"
 • I fill field "Username", "colobarator"
 • I fill field "Password", "123456"
 • I click "Sign In"
 • I see "Access for user collaborator is denied."
 ✓ OK in 2468ms
```

Note that scenarioContext can not be used in `Before/After` hooks due to Mocha limitation

### CODE REVIEW

File: lib/dataTable.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+/**
+ * Datatable class to provide data driven testing
+ */
+class dataTable {
```
Comment (DavertMik) [inline lib/dataTable.js]: classes start with capital: `DataTable`

File: lib/helper/Appium.js [hunk 2]
```diff
@@ -189,10 +189,7 @@ class Appium extends WebdriverIO {
   _after() {
     if (this.options.restart) return this.browser.end();
     if (this.isWeb && !this.platform) {
-      this.debugSection('Session', 'cleaning cookies and localStorage');
```
Comment (DavertMik) [inline lib/helper/Appium.js]: not related

Comment (APshenkin) [inline lib/helper/Appium.js]: @DavertMik yes, I created this branch from `appium-web-tests-improve` branch to have opportunity to test it locally with Appium helper.

File: lib/dataTable.js (hunk 1 above)
Comment (APshenkin) [inline lib/dataTable.js]: Absolutely right, I will rename it

**Pull request merged**